vdr/didx509/resolver_test.go

```diff
@@ -0,0 +1,514 @@
+package didx509
+
+import (
+	"crypto/sha1"
+	"crypto/sha512"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/minio/sha256-simd"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/pki"
+	resolver2 "github.com/nuts-foundation/nuts-node/vdr/resolver"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+	"strings"
+	"testing"
+)
+
+func TestManager_Resolve_OtherName(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	validator := pki.NewMockValidator(ctrl)
+	resolver := NewResolver(validator)
+	metadata := resolver2.ResolveMetadata{}
+
+	otherNameValue := "A_BIG_STRING"
+	_, certChain, rootCertificate, _, signingCert, err := BuildCertChain(otherNameValue)
+	require.NoError(t, err)
+	metadata.JwtProtectedHeaders = make(map[string]interface{})
+	metadata.JwtProtectedHeaders[X509CertChainHeader] = certChain
+	metadata.JwtProtectedHeaders[X509CertThumbprintHeader] = sha1Sum(signingCert.Raw)
+	metadata.JwtProtectedHeaders[X509CertThumbprintS256Header] = sha256Sum(signingCert.Raw)
+
+	rootDID := did.MustParseDID(fmt.Sprintf("did:x509:0:%s:%s::san:otherName:%s", "sha256", sha256Sum(rootCertificate.Raw), otherNameValue))
+
+	t.Run("test nulls", func(t *testing.T) {
+		chain, _ := metadata.GetProtectedHeaderChain(X509CertChainHeader)
+		delete(metadata.JwtProtectedHeaders, X509CertChainHeader)
+		_, _, err := resolver.Resolve(rootDID, &metadata)
+		require.Error(t, err)
+		assert.Equal(t, ErrX509ChainMissing.Error(), err.Error())
+		metadata.JwtProtectedHeaders[X509CertChainHeader] = chain
+
+	})
+	t.Run("test x5c cast issue", func(t *testing.T) {
+		chain, _ := metadata.GetProtectedHeaderChain(X509CertChainHeader)
+		metadata.JwtProtectedHeaders[X509CertChainHeader] = "GARBAGE"
+		_, _, err := resolver.Resolve(rootDID, &metadata)
+		require.Error(t, err)
+		assert.Equal(t, ErrX509ChainMissing.Error(), err.Error())
+		metadata.JwtProtectedHeaders[X509CertChainHeader] = chain
+
+	})
+	t.Run("happy flow, policy depth of 0", func(t *testing.T) {
+		rootDID := did.MustParseDID(fmt.Sprintf("did:x509:0:%s:%s", "sha256", sha256Sum(rootCertificate.Raw)))
+		validator.EXPECT().Validate(gomock.Any()).Return(nil)
+		resolve, documentMetadata, err := resolver.Resolve(rootDID, &metadata)
+
+		require.NoError(t, err)
+		assert.NotNil(t, resolve)
+		require.NoError(t, err)
+		assert.NotNil(t, documentMetadata)
+		// Check that the DID url is did#0
+		didUrl, err := did.ParseDIDURL(rootDID.String() + "#0")
+		assert.NotNil(t, resolve.VerificationMethod.FindByID(*didUrl))
+	})
+	t.Run("happy flow, policy depth of 1", func(t *testing.T) {
+		validator.EXPECT().Validate(gomock.Any()).Return(nil)
```

Comment:
FYI: mocks always return the default value if none is specified. So you don't need to add `.Return(nil)`, only if you actually want to return an error.

Reply:
Removed the .Return(nil)